utils/dists/MixtureExp.py

```diff
@@ -0,0 +1,58 @@
+"""
+Mixture of Exponential Distributions
+
+Author: Abhi (abhishek@myumanitoba.ca)
+"""
+import torch
+
+from torch import log, exp
+from torch.distributions.bernoulli import Bernoulli
+
+class MixtureExp(Bernoulli):
+    """
+    Mixture of overlapping exponential distributions from DVAE++
+    """
+    def __init__(self, logits=None, beta=None):
+        super(MixtureExp, self).__init__(logits=logits)
+        self.beta = beta
+        
+    def reparameterise(self):
+        """
+        - ICDF of mixture of exponential distributions (Eq. 3, DVAE++)
+        Returns:
+            zeta: approximate post samples
+        """
+        q = torch.sigmoid(self.logits)
+        q = torch.clamp(q, min=1e-7, max=1.-1e-7)
+        
+        rho = torch.rand(q.size())
+        b = (rho + torch.exp(-self.beta)*(q - rho))/((1. - q) - 1.)
```

Comment:
This is currently failing, as `self.beta` is an `int` but should be a `torch.int`. Probably just need to cast the type inline.